.github/workflows/bench.yml

```diff
@@ -0,0 +1,25 @@
+name: Benchmarks
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+
+jobs:
+  benchmarks:
```

Comment:
Why is this step taking 50% more time than our current huge test suite?

This seems off.

![diff](https://github.com/user-attachments/assets/6ce8970b-a466-4df2-b6a9-8baf225c0b4b)

Reply:
@maschad I share @arboleya concerns here. It is really strange why it is taking so long.

Could it be because of the codspeed setup process? Which involves downloading and installing it? https://github.com/FuelLabs/fuels-ts/actions/runs/10795148772/job/29941117888?pr=3084#step:5:37

Reply:
@Torres-ssf @arboleya that's certainly apart of the reason (probably in tandem with uploading that performance data as well) but I don't think there's any easy way around that looking at the [GH action](https://github.com/CodSpeedHQ/action)  

That's why I suggested that we try to optimize the CI setup and `pretest` elements since that's within our control, even if it's only for this job.

Reply:
The `bench:node` takes 5-6 secs locally for me.

Reply:
@maschad Is the Codspeed usage tied to the Github action `CodSpeedHQ/action@v3`? 

Is there no other way to use it to avoid the setup process?

Reply:
From the looks of it the [action essentially just installs](https://github.com/CodSpeedHQ/action/blob/main/action.yml#L43-L81)  the [runner](https://github.com/CodSpeedHQ/runner). Of course we need to use the runner to upload the results to Codspeed so I don't think avoiding the action would necessarily yield better results.

Looking at other runs across [other repos](https://github.com/prisma/prisma/actions/runs/10802495620/job/29964692421?pr=25209#step:7:42) it takes on average 9 - 12 minutes depending on the amount of benchmarks, so our 6 mins is less than others

You can look at more [here](https://codspeed.io/explore) 

We can optimize both our build times and the CodSpeed Github action, and even the runner if we determine it's worth the engineering effort.

Reply:
May be time to dig up #1753.

Reply:
@arboleya the issue is not with the installation script, that takes roughly ~20 seconds or less, as I had mentioned the bottleneck is [the runner](https://github.com/CodSpeedHQ/runner) which executes the vitest benchmarks in it's  [own environment](https://github.com/CodSpeedHQ/runner/blob/main/src/run/runner/run.rs#L18) and [measures the results](https://github.com/CodSpeedHQ/runner/blob/main/src/run/runner/valgrind.rs). This takes up 90%+ of the job time. 

I was looking to see if there was a way I could optimize it but to my dismay

![image](https://github.com/user-attachments/assets/45e77845-149f-431c-a951-e7449627c142)

Reply:
My bad; I misinterpreted what you said at first.

This is annoying, but I also don't want to hold progress.

If we exhaust all options, we can merge it as is and see how it goes. 🙂

Reply:
No worries @arboleya , I think your concern about the length of time this takes is very valid. For a performance tool, this is poor performance 😆 

We should definitely raise an issue on the runner to see if there is a way to optimize their execution flow